packages/terra-table/src/Table.jsx

```diff
@@ -547,6 +547,18 @@ function Table(props) {
     marginRight: hasResizableCol ? `${TableConstants.TABLE_MARGIN_RIGHT}px` : '0',
   };
 
+  // Set first and last row Ids
+  let firstRowId;
+  let lastRowId;
+
+  if (rows && rows.length) {
+    firstRowId = rows[0].id;
+    lastRowId = rows[rows.length - 1].id;
+  } else if (sections && sections.length && sections[0].rows && sections[0].rows.length) {
+    firstRowId = sections[0].rows[0].id;
+    lastRowId = sections[sections.length - 1].rows[sections[sections.length - 1].rows.length - 1].id;
+  }
```

Comment:
How is this logic dealing with the sections that are collapsed?  That logic existed in the clinical data grid code.

Reply:
Handled case of collapsible sections similar to `clinical-data-grid` where the visible first and last rows are determined to apply specific styling to cells: https://github.com/cerner/terra-framework/pull/1985/commits/fce4b663063a694bddaeaf64d6bd4ce240c1020f

https://github.com/cerner/terra-framework/pull/1985#discussion_r1483944966